R/date_standardization_helpers.R

```diff
@@ -13,32 +15,40 @@ date_check_timeframe <- function(first_date, last_date) {
   # make sure that they are single character strings in ISO 8601 format.
   iso_8601 <- "[0-9]{4}-(0|1(?=[0-2]))[0-9]-([0-2]|3(?=[0-1]))[0-9]"
 
+  # convert 'first_date' into Date in cases where it is provided as a
+  # character.
   first_date_is_character <- is.character(first_date)
-  first_date_has_len_1    <- length(first_date) == 1L
+  first_date_has_len_1 <- length(first_date) == 1L
   first_date_has_iso_8601 <- grepl(iso_8601, first_date, perl = TRUE)
   verdict <- first_date_is_character & first_date_has_len_1 &
     first_date_has_iso_8601
   if (verdict) {
     first_date <- as.Date(first_date, "%Y-%m-%d")
   }
 
+  # convert 'last_date' into Date in cases where it is provided as a
+  # character.
   last_date_is_character <- is.character(last_date)
-  last_date_has_len_1    <- length(last_date) == 1L
+  last_date_has_len_1 <- length(last_date) == 1L
   last_date_has_iso_8601 <- grepl(iso_8601, last_date, perl = TRUE)
   verdict <- last_date_is_character & last_date_has_len_1 &
     last_date_has_iso_8601
   if (verdict) {
     last_date <- as.Date(last_date, "%Y-%m-%d")
   }
 
-  # Set the first date to 50 years before the last date if it's not set
+  # When 'first_date' is not provided, set it to 50 years before the last date
   if (is.null(first_date) && inherits(last_date, "Date")) {
-    first_date <- min(seq.Date(last_date, length.out = 2L, by = "-50 years"))
+    first_date <- last_date - lubridate::years(50)
   }
 
+  # both 'first_date' and 'last_date' are expected to be of type Date
   if (!inherits(first_date, "Date") || !inherits(last_date, "Date")) {
-    stop("first_date and last_date must be Date objects or characters in",
-         "yyyy-mm-dd format.")
+    cli::cli_abort(c(
+      tr_("Unexpected format in the function arguments."),
+      i = tr_("{.emph first_date} and {.emph last_date} must be of type {.cls Date} or {.cls character} written in {.emph ISO8601} format ('2024-12-31' for December 31, 2024).") # nolint: line_length_linter
+    ))
+    stop("")
```

Comment:
Do you still need `stop()` if you're calling `cli_abort()`

Reply:
This is an ommission from me. I have deleted it.